src/main/webapp/app/exercises/programming/manage/update/programming-exercise-update.component.ts

```diff
@@ -276,8 +276,13 @@ export class ProgrammingExerciseUpdateComponent implements AfterViewInit, OnDest
             this.programmingExercise.maxStaticCodeAnalysisPenalty = undefined;
         }
 
-        // Automatically enable the checkout of the solution repository for Haskell exercises
-        this.programmingExercise.checkoutSolutionRepository = this.checkoutSolutionRepositoryAllowed && language === ProgrammingLanguage.HASKELL;
+        if (language == ProgrammingLanguage.HASKELL || language == ProgrammingLanguage.OCAML) {
+            // Instructors typically test against the example solution for Haskell and OCAML exercises, activate this option per default.
+            this.programmingExercise.checkoutSolutionRepository = this.checkoutSolutionRepositoryAllowed;
+        }
+        if (!this.checkoutSolutionRepositoryAllowed) {
+            this.programmingExercise.checkoutSolutionRepository = false;
+        }
```

Comment:
I think what confuses me the most is that you wrote "activate this option per default" and then set the value. Technically, the value can only be true at this point, but that only becomes clear once you dig deeper.
I assume, you do it as a failsafe in case the value changes in the future? I'd recommend then, making this clear in the comment.

Reply:
> I assume, you do it as a failsafe in case the value changes in the future?

Yes. As of right now, OCAML is only supported by LocalCI and it also supports checking out the solution repository. But there might me a case in the future where GitLab has a OCAML template, but does not allow checking out the solution repository.